mk/spksrc.service.installer

```diff
@@ -289,20 +296,32 @@ preupgrade ()
         syno_remove_user "${SYNOUSER_PREFIX}${USER}"
     fi
 
-    $MV "${TO_SAVE_DIR}/*" "$TMP_DIR" >> ${INST_LOG}
+    # Beware of /* outside the quotes, fails otherwise!
+    $MV "${INST_VAR}"/* "$TMP_DIR" >> ${INST_LOG}
     exit 0
 }
 
 postupgrade ()
 {
     log_step "postupgrade"
-    # Restore some stuff
-    $MV "$TMP_DIR" "$TO_SAVE_DIR" >> ${INST_LOG}
+
+    # Restore some stuff, has to be cp otherwise fails on directories
+    $CP "${TMP_DIR}"/* "${INST_VAR}" >> ${INST_LOG}
+
+    # Correct permissions of var folder
+    if [ $SYNOPKG_DSM_VERSION_MAJOR -lt 6 ]; then
```

Comment:
To avoid copy/paste from line 225, please create a new function `set_var_permissions`

Reply:
At line 225 the function uses a different folder for DSM5, while in the new part we use the same path for both DSM's.
```shell
        chown -R ${EFF_USER}:root "${SYNOPKG_PKGDEST}" >> ${INST_LOG} 2>&1
VS:
        chown -R ${EFF_USER}:${USER} "${INST_VAR}" >> ${INST_LOG} 2>&1
```

Reply:
So to have a `set_var_permissions` we would need to use the same path for both DSM versions in the things at line 225.